src/hotspot/cpu/s390/stubGenerator_s390.cpp

```diff
@@ -3053,6 +3053,29 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+  // load Method* target of MethodHandle
+  // Z_ARG1 = jobject receiver
+  // Z_method = Method* result
+  address generate_upcall_stub_load_target() {
+    StubCodeMark mark(this, "StubRoutines", "upcall_stub_load_target");
+    address start = __ pc();
+
+    __ resolve_jobject(Z_ARG1, Z_tmp_1, Z_tmp_2);
```

Comment:
@JornVernee is it possible to rebase to head stream ? If there are no issue on other archs :-) 

I have implemented `resolve_global_jobject` method for s390x and merged changes to head stream with PR https://github.com/openjdk/jdk/pull/20986. Once rebased you can push below change. 

```suggestion
    __ resolve_global_jobject(Z_ARG1, Z_tmp_1, Z_tmp_2);
```

Reply:
I've switched s390 to use `resolve_global_jobject` as well now.